react-state/react-state-quiz-notes.md

```diff
@@ -0,0 +1,50 @@
+# react-state-quiz-notes
+
+## Quiz Questions
+
+After completing this exercise, you should be able to discuss or answer the following questions:
+
+- What are _hooks_ in React?
+  special functions only found while React is rending. it allows you to hook into different react features.
+- What are the "Rules of Hooks"? (if necessary, re-read the "Pitfall" box in [State](https://react.dev/learn/state-a-components-memory))
+
+1. must start with "use"
+2. can only be called at top level of component
+3. cant call hooks inside nested functions, loops, or conditions
+
+- What is the purpose of state in React?
+  state is a built in object in react components that influences the behavior of a component and can be modified over time
+- Why can't we just maintain state in a local variable?
+  because local variables cant trigger rerendering when its value changes
```

Comment:
doesnt update on screen, just changes value. use state setter to rerender component if there is a change